### PR TITLE
feature(planner): Support some scalar expressions in new planner

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- Support `TRY_CAST`
- Support `EXTRACT`
- Support `INTERVAL` literal
- Fix unary operators

## Changelog

- New Feature

### CODE REVIEW

File: query/src/sql/planner/semantic/type_check.rs [hunk 1]
```diff
@@ -438,10 +469,65 @@ impl<'a> TypeChecker<'a> {
...
     ) -> Result<(Scalar, DataTypeImpl)> {
-        self.resolve_function(op.to_string().as_str(), &[child], required_type)
+        if *op == UnaryOperator::Plus {
+            // Omit unary + operator
+            return self.resolve(child, required_type).await;
+        }
+        let op_name = match op {
+            UnaryOperator::Minus => "negate".to_string(),
+            _ => op.to_string(),
+        };
```
Comment (andylokandy) [inline query/src/sql/planner/semantic/type_check.rs:479]: Explicitly match against `Plus` and `Not`?

Comment (leiysky) [reply query/src/sql/planner/semantic/type_check.rs]: Done.

File: query/src/sql/planner/semantic/type_check.rs [hunk 2]
```diff
@@ -473,22 +559,44 @@ impl<'a> TypeChecker<'a> {
...
+                DateTimeField::Minute => IntervalType::new_impl(IntervalKind::Minute),
+                DateTimeField::Second => IntervalType::new_impl(IntervalKind::Second),
+                _ => {
+                    return Err(ErrorCode::SemanticError(format!(
+                        "Invalid interval kind: {}",
+                        interval.field
+                    )));
+                }
+            };
+        }
```
[the remark this answers is not in the document]
Comment (leiysky) [reply query/src/sql/planner/semantic/type_check.rs]: Done

**Review by andylokandy — APPROVED** [review]
rest LGTM

File: query/src/sql/planner/semantic/type_check.rs [hunk 3]
```diff
@@ -438,8 +469,68 @@ impl<'a> TypeChecker<'a> {
...
+                self.resolve(child, required_type).await
+            }
+
+            UnaryOperator::Minus => {
+                self.resolve_function("negate", &[child], required_type)
+                    .await
+            }
+
+            _ => {
+                self.resolve_function(op.to_string().as_str(), &[child], required_type)
```
[the remark this answers is not in the document]
Comment (leiysky) [reply query/src/sql/planner/semantic/type_check.rs:484]: Done.

File: query/src/sql/planner/semantic/type_check.rs [hunk 4]
```diff
@@ -438,8 +469,68 @@ impl<'a> TypeChecker<'a> {
...
+                self.resolve(child, required_type).await
+            }
+
+            UnaryOperator::Minus => {
+                self.resolve_function("negate", &[child], required_type)
+                    .await
+            }
+
+            UnaryOperator::Not => {
+                self.resolve_function(op.to_string().as_str(), &[child], required_type)
```
Comment (andylokandy) [inline query/src/sql/planner/semantic/type_check.rs:484]:
```suggestion
                self.resolve_function("not", &[child], required_type)
```

**Pull request merged**